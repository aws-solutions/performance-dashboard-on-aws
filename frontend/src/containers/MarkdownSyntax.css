a {
  color: #565c65;
}

a:hover {
  color: #1b1b1b;
}

a:active {
  color: #1b1b1b;
}

a:visited {
  color: #1b1b1b;
}
