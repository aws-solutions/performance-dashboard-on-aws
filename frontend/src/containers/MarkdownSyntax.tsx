/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import MarkdownRender from "../components/MarkdownRender";
import { useTranslation } from "react-i18next";

const MarkdownSyntax = () => {
    const { t } = useTranslation();

    return (
        <>
            <h1>{t("MarkdownSyntax.Label")}</h1>
            <p className="font-sans-lg">{t("MarkdownSyntax.Description")}</p>

            <h2 className="margin-top-5">{t("MarkdownSyntax.Bold")}</h2>
            <p>{t("MarkdownSyntax.BoldDescription")}</p>
            <table className="usa-table usa-table--borderless" width="100%">
                <thead>
                    <tr>
                        <th style={{ width: "50%" }}>{t("MarkdownSyntax.Markdown")}</th>
                        <th style={{ width: "50%" }}>{t("MarkdownSyntax.RenderedOutput")}</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <td>{t("MarkdownSyntax.ILoveBoldText")}</td>
                        <td>
                            <MarkdownRender source={t("MarkdownSyntax.ILoveBoldText")} />
                        </td>
                    </tr>
                </tbody>
            </table>

            <h2 className="margin-top-7">{t("MarkdownSyntax.Hyperlink")}</h2>
            <p>{t("MarkdownSyntax.HyperlinkDescription")}</p>
            <table className="usa-table usa-table--borderless" width="100%">
                <thead>
                    <tr>
                        <th style={{ width: "50%" }}>{t("MarkdownSyntax.Markdown")}</th>
                        <th style={{ width: "50%" }}>{t("MarkdownSyntax.RenderedOutput")}</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <td>{t("MarkdownSyntax.ILoveAWS")}</td>
                        <td>
                            <MarkdownRender source={t("MarkdownSyntax.ILoveAWS")} />
                        </td>
                    </tr>
                </tbody>
            </table>

            <h2 className="margin-top-7">{t("MarkdownSyntax.UnorderedList")}</h2>
            <p>{t("MarkdownSyntax.UnorderedListDescription")}</p>
            <table className="usa-table usa-table--borderless" width="100%">
                <thead>
                    <tr>
                        <th style={{ width: "50%" }}>{t("MarkdownSyntax.Markdown")}</th>
                        <th style={{ width: "50%" }}>{t("MarkdownSyntax.RenderedOutput")}</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <td>
                            <div>{t("MarkdownSyntax.FirstItem")}</div>
                            <div>{t("MarkdownSyntax.SecondItem")}</div>
                            <div>{t("MarkdownSyntax.ThirdItem")}</div>
                            <div>{t("MarkdownSyntax.FourthItem")}</div>
                        </td>
                        <td>
                            <MarkdownRender
                                source={`${t("MarkdownSyntax.FirstItem")}
${t("MarkdownSyntax.SecondItem")}
${t("MarkdownSyntax.ThirdItem")}
${t("MarkdownSyntax.FourthItem")}`}
                            />
                        </td>
                    </tr>
                </tbody>
            </table>
        </>
    );
};

export default MarkdownSyntax;
