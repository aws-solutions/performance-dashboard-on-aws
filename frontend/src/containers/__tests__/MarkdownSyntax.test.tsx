/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render } from "@testing-library/react";
import MarkdownSyntax from "../MarkdownSyntax";
import { MemoryRouter } from "react-router-dom";

test("renders the markdown syntax component", async () => {
    const wrapper = render(<MarkdownSyntax />);
    expect(wrapper.container).toMatchSnapshot();
});

test("renders the title of the markdown syntax component", async () => {
    const { findByText } = render(<MarkdownSyntax />, {
        wrapper: MemoryRouter,
    });
    const title = await findByText("Markdown Syntax");
    expect(title).toBeInTheDocument();
});

test("renders the description of the markdown syntax component", async () => {
    const { findByText } = render(<MarkdownSyntax />, {
        wrapper: MemoryRouter,
    });
    const title = await findByText(
        "Supported Markdown is limited to bold, hyperlinks and " +
            "single-level unordered lists. All other text and markdown " +
            "will be rendered as plain text.",
    );
    expect(title).toBeInTheDocument();
});

test("renders the Bold of the markdown syntax component", async () => {
    const { findByText } = render(<MarkdownSyntax />, {
        wrapper: MemoryRouter,
    });
    const title = await findByText("Bold");
    expect(title).toBeInTheDocument();
});

test("renders the Bold description of the markdown syntax component", async () => {
    const { findByText } = render(<MarkdownSyntax />, {
        wrapper: MemoryRouter,
    });
    const title = await findByText(
        "To bold text, add two asterisks or underscores before and after a word " +
            "or phrase. To bold the middle of a word for emphasis, add two asterisks " +
            "without spaces around the letters.",
    );
    expect(title).toBeInTheDocument();
});

test("renders the Hyperlink title of the markdown syntax component", async () => {
    const { findByText } = render(<MarkdownSyntax />, {
        wrapper: MemoryRouter,
    });
    const title = await findByText("Hyperlink");
    expect(title).toBeInTheDocument();
});

test("renders the Hyperlink description of the markdown syntax component", async () => {
    const { findByText } = render(<MarkdownSyntax />, {
        wrapper: MemoryRouter,
    });
    const title = await findByText(
        "To create a link, enclose the link text in brackets (e.g., [AWS]) and " +
            "then follow it immediately with the URL in parentheses (e.g., " +
            "(https://aws.amazon.com)).",
    );
    expect(title).toBeInTheDocument();
});

test("renders the Unordered list title of the markdown syntax component", async () => {
    const { findByText } = render(<MarkdownSyntax />, {
        wrapper: MemoryRouter,
    });
    const title = await findByText("Unordered list");
    expect(title).toBeInTheDocument();
});

test("renders the Unordered list description of the markdown syntax component", async () => {
    const { findByText } = render(<MarkdownSyntax />, {
        wrapper: MemoryRouter,
    });
    const title = await findByText(
        "To create an unordered list, add dashes (-), asterisks (*), or plus " +
            "signs (+) and space in front of line items.",
    );
    expect(title).toBeInTheDocument();
});
