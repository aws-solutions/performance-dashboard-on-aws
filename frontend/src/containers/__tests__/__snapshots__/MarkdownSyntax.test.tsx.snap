// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the markdown syntax component 1`] = `
<div>
  <h1>
    Markdown Syntax
  </h1>
  <p
    class="font-sans-lg"
  >
    Supported Markdown is limited to bold, hyperlinks and single-level unordered lists. All other text and markdown will be rendered as plain text.
  </p>
  <h2
    class="margin-top-5"
  >
    Bold
  </h2>
  <p>
    To bold text, add two asterisks or underscores before and after a word or phrase. To bold the middle of a word for emphasis, add two asterisks without spaces around the letters.
  </p>
  <table
    class="usa-table usa-table--borderless"
    width="100%"
  >
    <thead>
      <tr>
        <th
          style="width: 50%;"
        >
          Markdown
        </th>
        <th
          style="width: 50%;"
        >
          Rendered output
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          I love **bold** text
        </td>
        <td>
          <div
            class="Markdown undefined"
          >
            <p>
              I love 
              <strong>
                bold
              </strong>
               text
            </p>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
  <h2
    class="margin-top-7"
  >
    Hyperlink
  </h2>
  <p>
    To create a link, enclose the link text in brackets (e.g., [AWS]) and then follow it immediately with the URL in parentheses (e.g., (https://aws.amazon.com)).
  </p>
  <table
    class="usa-table usa-table--borderless"
    width="100%"
  >
    <thead>
      <tr>
        <th
          style="width: 50%;"
        >
          Markdown
        </th>
        <th
          style="width: 50%;"
        >
          Rendered output
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          I love [AWS](https://aws.amazon.com)
        </td>
        <td>
          <div
            class="Markdown undefined"
          >
            <p>
              I love 
              <a
                href="https://aws.amazon.com"
                target="_blank"
              >
                AWS
              </a>
            </p>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
  <h2
    class="margin-top-7"
  >
    Unordered list
  </h2>
  <p>
    To create an unordered list, add dashes (-), asterisks (*), or plus signs (+) and space in front of line items.
  </p>
  <table
    class="usa-table usa-table--borderless"
    width="100%"
  >
    <thead>
      <tr>
        <th
          style="width: 50%;"
        >
          Markdown
        </th>
        <th
          style="width: 50%;"
        >
          Rendered output
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <div>
            + First item
          </div>
          <div>
            * Second item
          </div>
          <div>
            - Third item
          </div>
          <div>
            + Fourth item
          </div>
        </td>
        <td>
          <div
            class="Markdown undefined"
          >
            <ul>
              <li>
                First item
              </li>
            </ul>
            <ul>
              <li>
                Second item
              </li>
            </ul>
            <ul>
              <li>
                Third item
              </li>
            </ul>
            <ul>
              <li>
                Fourth item
              </li>
            </ul>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
